src/Nethermind/Nethermind.Synchronization/Peers/PeerInfo.cs

```diff
@@ -58,7 +58,8 @@ public PeerInfo(ISyncPeer syncPeer)
         public bool CanBeAllocated(AllocationContexts contexts)
         {
             return !IsAsleep(contexts) &&
-                   !IsAllocated(contexts);
+                   !IsAllocated(contexts) &&
+                   this.SupportsAllocation(contexts);
```

Comment:
[You can ignore it] Maybe a better method name would be IsSupported or HasSupportForAllocation?